### PR TITLE
9076 - Added routes and empty pages for STLT dashboard

### LINKED ISSUE #9076: Add stub pages: main index, facilities/providers index, facilities/providers show, report details
## Problem statement

We need to add route scaffolding to support the STLT Dashboard pages.

## What you need to know

There are four types of pages:
- Main index page
- "All facilities & providers" page
- Facility/provider/submitter details page
- Report details page

## Acceptance criteria

- [x] Routes are added for each page type
- [x] Empty pages are added for each route

### PR DESCRIPTION
**Pull request by penny-lischer**
This PR ...

Added routes and empty pages for STLT dashboard

Test Steps:
1. Log into RS
2. You can now view

## Changes
Added the following routes:
/data-dashboard
/data-dashboard/facilities-providers
/data-dashboard/facilities-providers/:facilityProviderSubmitter
/data-dashboard/report-details/:reportId

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [x] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

## Linked Issues
- Fixes #9076

### CODE REVIEW

**Comment by etanb** [comment]
@penny-lischer are we checking in those spare empty pages? can we wait to merge this until there's content and just merge related changes into this branch?

File: frontend-react/src/components/header/ReportStreamHeader.tsx [hunk 1]
```diff
@@ -62,6 +62,13 @@ export const ReportStreamHeader = () => {
                 >
                     <span>{FeatureName.DAILY_DATA}</span>
                 </USNavLink>
+                // <USNavLink
```
[the remark this answers is not in the document]
Comment (penny-lischer) [reply frontend-react/src/components/header/ReportStreamHeader.tsx:65]: @etanb this will be uncommented in a future story. Was intentional

**Comment by penny-lischer** [comment]
> @penny-lischer are we checking in those spare empty pages? can we wait to merge this until there's content and just merge related changes into this branch?

This pages will be worked on in this sprint and next. They are placeholders for now so that more than one dev can work on the pages in parallel.

File: frontend-react/src/components/header/ReportStreamHeader.tsx (hunk 1 above)
Comment (jpandersen87) [inline frontend-react/src/components/header/ReportStreamHeader.tsx:65]: Totally OK if intended for the future, just leave a note on it to indicate this.

**Review by jpandersen87 — CHANGES_REQUESTED** [review]
You just need to leave a note on the commented out code and you're good.

**Review by jpandersen87 — APPROVED** [review]

File: frontend-react/src/components/header/ReportStreamHeader.tsx [hunk 2]
```diff
@@ -62,6 +62,14 @@ export const ReportStreamHeader = () => {
                 >
                     <span>{FeatureName.DAILY_DATA}</span>
                 </USNavLink>
+                // Will be needed when all STLT pages are developed
```
Comment (etanb) [inline frontend-react/src/components/header/ReportStreamHeader.tsx:65]: Can we leave out any commented code until we need it? Any commented code in a `render` appears in the DOM.

Comment (jpandersen87) [reply frontend-react/src/components/header/ReportStreamHeader.tsx:65]: I was under the impression, for at least the production build, comments are stripped from the final bundle. Is this not the case?

Comment (etanb) [reply frontend-react/src/components/header/ReportStreamHeader.tsx:65]: I haven't seen any evidence of that, but I can run a production build locally to text. The only solution I've seen recommended is: https://www.npmjs.com/package/webpack-comment-remover-loader

In either case, we shouldn't have comments within our `render` functions. I think comments within the logic beforehand is fine for complicated business logic.

Comment (jpandersen87) [reply frontend-react/src/components/header/ReportStreamHeader.tsx:65]: I see now. To be clear (and I agree), there should not be comments within JSX blocks (which are rendered). Easy fix, just move the comments to above the `itemsMenu.push(` (within the logic side).

**Review by etanb — APPROVED** [review]

**Pull request merged**